File: api/gen_qrcode.py
import requests, json
from http.server import BaseHTTPRequestHandler

CORP_ID = "ww2495c9000fdccc32"
SECRET  = "yi2NG5s5Mbx79YSOTJOuv6XKh8oF9Z060picBsCBLAo"

def get_token():
    r = requests.get(f"https://qyapi.weixin.qq.com/cgi-bin/gettoken?corpid={CORP_ID}&corpsecret={SECRET}").json()
    return r["access_token"]

class handler(BaseHTTPRequestHandler):
    def do_POST(self):
        length       = int(self.headers.get("Content-Length", 0))
        body         = json.loads(self.rfile.read(length))
        chat_id      = body.get("chat_id", "")
        partner_name = body.get("partner_name", "新伙伴")

        token = get_token()
        resp  = requests.post(
            f"https://qyapi.weixin.qq.com/cgi-bin/externalcontact/groupchat/add_join_way?access_token={token}",
            json={
                "scene": 1,
                "remark": partner_name,
                "auto_create_room": 0,
                "chat_id_list": [chat_id]
            }
        ).json()

        if resp.get("errcode") == 0:
            config_id = resp["config_id"]
            result = {"ok": True, "qr_url": f"https://work.weixin.qq.com/gm/{config_id}"}
        else:
            result = {"ok": False, "error": resp.get("errmsg")}

        self.send_response(200)
        self.send_header("Content-Type", "application/json")
        self.send_header("Access-Control-Allow-Origin", "*")
        self.end_headers()
        self.wfile.write(json.dumps(result).encode())

    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header("Access-Control-Allow-Origin", "*")
        self.send_header("Access-Control-Allow-Methods", "POST, OPTIONS")
        self.send_header("Access-Control-Allow-Headers", "Content-Type")
        self.end_headers()

    def log_message(self, *args):
        pass

File: api/gen_qrcode.js
const https = require('https');

const CORP_ID = 'ww2495c9000fdccc32';
const SECRET  = 'yi2NG5s5Mbx79YSOTJOuv6XKh8oF9Z060picBsCBLAo';

function get(url) {
  return new Promise((resolve, reject) => {
    https.get(url, res => {
      let d = '';
      res.on('data', c => d += c);
      res.on('end', () => resolve(JSON.parse(d)));
    }).on('error', reject);
  });
}

function post(url, body) {
  return new Promise((resolve, reject) => {
    const data = JSON.stringify(body);
    const u = new URL(url);
    const req = https.request({
      hostname: u.hostname, path: u.pathname + u.search,
      method: 'POST',
      headers: { 'Content-Type': 'application/json', 'Content-Length': Buffer.byteLength(data) }
    }, res => {
      let d = '';
      res.on('data', c => d += c);
      res.on('end', () => resolve(JSON.parse(d)));
    });
    req.on('error', reject);
    req.write(data);
    req.end();
  });
}

module.exports = async (req, res) => {
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') { res.status(200).end(); return; }

  const { chat_id, partner_name = '新伙伴' } = req.body;

  const { access_token } = await get(
    `https://qyapi.weixin.qq.com/cgi-bin/gettoken?corpid=${CORP_ID}&corpsecret=${SECRET}`
  );

  const resp = await post(
    `https://qyapi.weixin.qq.com/cgi-bin/externalcontact/groupchat/add_join_way?access_token=${access_token}`,
    { scene: 1, remark: partner_name, auto_create_room: 0, chat_id_list: [chat_id] }
  );

  if (resp.errcode === 0) {
    res.json({ ok: true, qr_url: `https://work.weixin.qq.com/gm/${resp.config_id}` });
  } else {
    res.json({ ok: false, error: resp.errmsg });
  }
};
